edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/KubernetesEnvironmentOperatorTest.cs

```diff
@@ -0,0 +1,150 @@
+        }
+
+        [Fact]
+        public async Task DoesNotCollectModuleRuntimeInfoForUnknownModules()
+        {
+            var tokenSource = new CancellationTokenSource(DefaultTimeout);
+
+            var labels = new Dictionary<string, string> { ["a"] = "b" };
+            await this.client.AddModuleServiceAccountAsync("foreign-pod", labels, null);
+            await this.client.AddModuleDeploymentAsync("foreign-pod", labels, null);
+            await this.client.WaitUntilAnyPodsAsync("status.phase=Running", tokenSource.Token);
+
+            ModuleRuntimeInfo moduleInfo = await this.GetModule("foreign-pod");
+
+            moduleInfo.Should().BeNull();
+        }
+
+        [Fact]
+        public async Task DeletesModuleRuntimeInfoForDeletedModules()
+        {
+            var tokenSource = new CancellationTokenSource(DefaultTimeout * 3);
+
+            await this.AddEdgeModule("Module-A");
+            await this.client.WaitUntilAnyPodsAsync("status.phase=Running", tokenSource.Token);
+
+            IEnumerable<ModuleRuntimeInfo> modules = await this.runtimeInfoProvider.GetModules(CancellationToken.None);
+            modules.Should().HaveCount(1);
+
+            await this.client.DeleteModuleDeploymentAsync("module-a");
+            await this.client.WaitUntilPodsExactNumberAsync(0, tokenSource.Token);
+
+            modules = await this.runtimeInfoProvider.GetModules(CancellationToken.None);
+            modules.Should().BeEmpty();
+        }
+
+        [Fact(Skip = "Reveals Bug 5576942")]
+        public async Task UpdatesModuleRuntimeInfoWhenModuleDeploymentUpdated()
+        {
+            var tokenSource = new CancellationTokenSource(DefaultTimeout * 3);
+
+            await this.AddEdgeModule("Module-A");
+            await this.client.WaitUntilAnyPodsAsync("status.phase=Running", tokenSource.Token);
+            ModuleRuntimeInfo initialModuleInfo = await this.GetModule("Module-A");
+
+            await this.client.ReplaceModuleImageAsync("module-a", "alpine:latest");
+            await this.client.WaitUntilPodsExactNumberAsync(2, tokenSource.Token);
+            await this.client.WaitUntilPodsExactNumberAsync(1, tokenSource.Token);
+            ModuleRuntimeInfo updatedModuleInfo = await this.GetModule("Module-A");
+
+            initialModuleInfo.Should().NotBeNull();
+            initialModuleInfo.StartTime.Should().NotBeNone();
+            initialModuleInfo.ModuleStatus.Should().Be(ModuleStatus.Running);
+
+            updatedModuleInfo.Should().NotBeNull();
+            updatedModuleInfo.StartTime.Should().NotBeNone();
+            initialModuleInfo.ModuleStatus.Should().Be(ModuleStatus.Running);
+            updatedModuleInfo.StartTime.OrDefault().Should().BeAfter(initialModuleInfo.StartTime.OrDefault());
+        }
+
+        async Task<ModuleRuntimeInfo> GetModule(string name)
+        {
+            IEnumerable<ModuleRuntimeInfo> modules = await this.runtimeInfoProvider.GetModules(CancellationToken.None);
```

Comment:
I am finding the use of a common runtimeInfoProvider a little precarious because one test failure can affect the other. It would be much easier to consume test results when each test sets up modules indepedently of the other.